trinity/utils/ipc.py

```diff
@@ -14,33 +14,39 @@ def wait_for_ipc(ipc_path: str, timeout: int=1) -> None:
 
 
 def kill_process_gracefully(process: Process,
-                            logger: Logger=None,
+                            logger: Logger,
```

Comment:
Thoughts on just using a local logger for this?  Reason to send in an outside logger?

Reply:
Not sure; I've just made the logger required to ensure we always log what goes on here, since that is important. Before this change `trinity.main()` was not passing a logger so it was all lost